zowe/zowe-explorer-vscode
# feat: Expose local storage through access control facility

## Proposed changes

Resolves #3180 

**Dev doc:** https://github.com/zowe/zowe-explorer-vscode/wiki/Local-Storage

### How to test

- Build a VSIX for Zowe Explorer using this PR and install it in VS Code
- Checkout the `local-storage-sample` branch on https://github.com/zowe/zowe-client-samples
- `cd vscode-extension-samples/local-storage-sample && npm install && npm run compile`
- After the sample build has finished, click on the Run & Debug panel on the VS Code Side Bar
- Select "Run Extension" from the tasks list at the top and select the Play ▶ button
- Notice the notifications for the sample extension:
  - The first notification shown is for the list of writable keys
  - 2nd notification: list of readable keys
  - 3rd notification: the value for `zowe.ds.history` (a readable key)
  - 4th notification: trying to access a key that isn't exposed

## Release Notes

Milestone: 3.1.0

Changelog:

- Exposed read and write access to local storage keys for Zowe Explorer extenders. [#3180](https://github.com/zowe/zowe-explorer-vscode/issues/3180)

## Types of changes



  - Fixed issue where workspace settings were merged into wrong local storage layer during migration
  - Added support for `ZoweLocalStorage` to save in a workspace
  - Added the local storage access facility for extenders

## Checklist



**General**

  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**


**Deployment**

  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
